### ISSUE TITLE
Cannot parse STIX 1.1.1 files which is downloaded via /events/restSearch.

### ISSUE THREAD

**Issue by stmtstk** [MISP/MISP#7363]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu 20.04.1 LTS (GNU/Linux 5.8.0-36-generic x86_64)
| OS version (client)       | Ubuntu 20.04.1 LTS (GNU/Linux 5.8.0-36-generic x86_64)
| PHP version               | PHP 7.4.3
| MISP version / git hash   | 2.4.135
| Browser                   | Google chrome

### Expected behavior

STIX 1.1.1 files exported from MISP are parsed successfully by python-misp library.

### Actual behavior

Parsing failed.
However, when I added the namespace phrase below manually, the files parsed successfully.

```
        xmlns:stix-ciqidentity="http://stix.mitre.org/extensions/Identity#CIQIdentity3.0-1"
```

### Steps to reproduce the behavior

1. Create a new event
2. Add a target-user attribute in the event and fill any value.
3. Click "Download files" and choose "STIX XML".
4. We can confirm that the downloaded STIX files luck the "xmlns:stix-ciqidentity" namespace.

### Logs, screenshots, configuration dump, ...
N/A

**Comment by chrisinmtown** [comment]
FWIW I tested this scenario on a 2.4.133 anad 2.4.141 system, using the web UI to view an event then Download as.., STIX 1 XML. I see the same result on both, an XML file with a root element "STIX_Package" with this xmlns decl on it:
```
	xmlns:ciqIdentity="http://stix.mitre.org/extensions/Identity#CIQIdentity3.0-1"
```
But I do *not* see a decl "xmlns:stix-ciqIdentity". Further, no element in my example event download XML files uses the ciqIdentity namespace, but maybe I just chose events that doesn't have data which requires that namespace.

**Comment by stmtstk** [comment]
```
                        <incident:Victim id="ORGNAME:Identity-9ac8f724-0fb2-4aaa-8a89-24322eeb3ca3" xsi:type="stix-ciqidentity:CIQIdentity3.0InstanceType">
                            <stixCommon:Name>Targeting data: test target user (MISP Attribute)</stixCommon:Name>
                            <stix-ciqidentity:Specification xmlns:stix-ciqidentity="http://stix.mitre.org/extensions/Identity#CIQIdentity3.0-1">
              <xpil:PartyName xmlns:xpil="urn:oasis:names:tc:ciq:xpil:3">
                <xnl:PersonName xmlns:xnl="urn:oasis:names:tc:ciq:xnl:3">
                  <xnl:NameElement>test target user</xnl:NameElement>
                </xnl:PersonName>
              </xpil:PartyName>
            </stix-ciqidentity:Specification>
                        </incident:Victim>
```
In the STIX 1.1.1 file which is converted by MISP contains `<incident:Victim>` block.
The `xsi:type` of `<incident:Victim>` starts with `stix-ciqidentity` instead of `ciqIdentity`.
It seems to be a reason to fail parsing.

**Comment by chrisinmtown** [comment]
I agree that if an element uses a namespace that's not defined in the document it will not parse. I checked the Stix source and I definitely see the stix-ciqIdentity namespace being defined there:
```
./extensions/identity/ciq_identity_3_0.py:et.register_namespace('stix-ciqidentity', XML_NS_STIX_EXT)
./extensions/identity/ciq_identity_3_0.py:    _XML_NS_PREFIX = "stix-ciqidentity"
./extensions/identity/ciq_identity_3_0.py:    _XSI_TYPE = "stix-ciqidentity:CIQIdentity3.0InstanceType"
```
In the example you show, this element defines *and* uses the stix-ciqidentity namespace; you are saying you had to add that definition on that element, is that right? 
```
<stix-ciqidentity:Specification xmlns:stix-ciqidentity="http://stix.mitre.org/extensions/Identity#CIQIdentity3.0-1">
```

**Comment by stmtstk** [comment]
Thanks @chrisinmtown 

The namespaces in the `STIXPackage` is below.

```
<stix:STIX_Package 
	xmlns:cyboxCommon="http://cybox.mitre.org/common-2"
	xmlns:cybox="http://cybox.mitre.org/cybox-2"
	xmlns:cyboxVocabs="http://cybox.mitre.org/default_vocabularies-2"
	xmlns:AccountObj="http://cybox.mitre.org/objects#AccountObject-2"
	xmlns:ArtifactObj="http://cybox.mitre.org/objects#ArtifactObject-2"
	xmlns:ASObj="http://cybox.mitre.org/objects#ASObject-1"
	xmlns:AddressObj="http://cybox.mitre.org/objects#AddressObject-2"
	xmlns:PortObj="http://cybox.mitre.org/objects#PortObject-2"
	xmlns:DomainNameObj="http://cybox.mitre.org/objects#DomainNameObject-1"
	xmlns:EmailMessageObj="http://cybox.mitre.org/objects#EmailMessageObject-2"
	xmlns:FileObj="http://cybox.mitre.org/objects#FileObject-2"
	xmlns:HTTPSessionObj="http://cybox.mitre.org/objects#HTTPSessionObject-2"
	xmlns:HostnameObj="http://cybox.mitre.org/objects#HostnameObject-1"
	xmlns:MutexObj="http://cybox.mitre.org/objects#MutexObject-2"
	xmlns:PipeObj="http://cybox.mitre.org/objects#PipeObject-2"
	xmlns:URIObj="http://cybox.mitre.org/objects#URIObject-2"
	xmlns:WinRegistryKeyObj="http://cybox.mitre.org/objects#WinRegistryKeyObject-2"
	xmlns:WinServiceObj="http://cybox.mitre.org/objects#WinServiceObject-2"
	xmlns:NetworkConnectionObj="http://cybox.mitre.org/objects#NetworkConnectionObject-2"
	xmlns:NetworkSocketObj="http://cybox.mitre.org/objects#NetworkSocketObject-2"
	xmlns:SocketAddressObj="http://cybox.mitre.org/objects#SocketAddressObject-1"
	xmlns:SystemObj="http://cybox.mitre.org/objects#SystemObject-2"
	xmlns:ProcessObj="http://cybox.mitre.org/objects#ProcessObject-2"
	xmlns:X509CertificateObj="http://cybox.mitre.org/objects#X509CertificateObject-2"
	xmlns:WhoisObj="http://cybox.mitre.org/objects#WhoisObject-2"
	xmlns:WinExecutableFileObj="http://cybox.mitre.org/objects#WinExecutableFileObject-2"
	xmlns:UnixUserAccountObj="http://cybox.mitre.org/objects#UnixUserAccountObject-2"
	xmlns:UserAccountObj="http://cybox.mitre.org/objects#UserAccountObject-2"
	xmlns:WinUserAccountObj="http://cybox.mitre.org/objects#WinUserAccountObject-2"
	xmlns:CustomObj="http://cybox.mitre.org/objects#CustomObject-1"
	xmlns:marking="http://data-marking.mitre.org/Marking-1"
	xmlns:simpleMarking="http://data-marking.mitre.org/extensions/MarkingStructure#Simple-1"
	xmlns:tlpMarking="http://data-marking.mitre.org/extensions/MarkingStructure#TLP-1"
	xmlns:et="http://stix.mitre.org/ExploitTarget-1"
	xmlns:incident="http://stix.mitre.org/Incident-1"
	xmlns:indicator="http://stix.mitre.org/Indicator-2"
	xmlns:coa="http://stix.mitre.org/CourseOfAction-1"
	xmlns:ttp="http://stix.mitre.org/TTP-1"
	xmlns:ta="http://stix.mitre.org/ThreatActor-1"
	xmlns:stixCommon="http://stix.mitre.org/common-1"
	xmlns:stixVocabs="http://stix.mitre.org/default_vocabularies-1"
	xmlns:ciqIdentity="http://stix.mitre.org/extensions/Identity#CIQIdentity3.0-1"
	xmlns:snortTM="http://stix.mitre.org/extensions/TestMechanism#Snort-1"
	xmlns:stix="http://stix.mitre.org/stix-1"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:xal="urn:oasis:names:tc:ciq:xal:3"
	xmlns:xnl="urn:oasis:names:tc:ciq:xnl:3"
	xmlns:xpil="urn:oasis:names:tc:ciq:xpil:3"
	xmlns:ORGNAME="https://10.0.3.108"
	xmlns:xlink="http://www.w3.org/1999/xlink"
	xmlns:ds="http://www.w3.org/2000/09/xmldsig#"
	xmlns:xs="http://www.w3.org/2001/XMLSchema"
	 id="ORGNAME:Package-f22746ca-a2ad-4f1a-bf9b-b8d18fea81cf" version="1.1.1" timestamp="2021-04-22T10:46:15.728604">
```

This tag does not contain `stix-ciqidentity`.
So I manually inserted the phrase below.
```
xmlns:stix-ciqidentity="http://stix.mitre.org/extensions/Identity#CIQIdentity3.0-1"
```

> In the example you show, this element defines and uses the stix-ciqidentity namespace; you are saying you had to add that definition on that element, is that right?

Yes. 
The STIX files were parsed successfully after I added the `stix-ciqidentity` namespace in the downloaded STIX files.

**Comment by chrisinmtown** [comment]
Maybe this is a dumb question, but how do you parse XML files with the python-misp (pymisp?) library?

**Comment by stmtstk** [comment]
@chrisinmtown 

```
from stix.core.stix_package import STIXPackage

STIXPackage.from_xml('<<<file path of downloaded stix 1.1.1>>>')
```

I used `STIXPackage.from_xml()` to parse them.

**Comment by chrisr3d** [comment]
Hey do you have a sample that I could use to reproduce the issue on my side?

**Comment by stmtstk** [comment]
Try below.

### Steps to reproduce the behavior

1. Create a new event
2. Add a target-user attribute in the event and fill any value.
3. Click "Download files" and choose "STIX XML".
4. We can confirm that the downloaded STIX files luck the "xmlns:stix-ciqidentity" namespace.

**Comment by chrisr3d** [comment]
Right, I see the corresponding namespace was wrong.

It should be fixed with 6347b24d2, let me know if it is not the case and reopen the issue if needed.

**Comment by stmtstk** [comment]
I think it seems to be good. Thank you!
